internal/host/validator.go

```diff
@@ -343,38 +343,38 @@ func (v *validator) printCompatibleWithClusterPlatform(c *validationContext, sta
 	}
 }
 
-func isDiskEncryptionRelevantForHost(c *validationContext) bool {
-	switch *c.cluster.DiskEncryption.EnableOn {
+func isDiskEncryptionEnabledForRole(encryption models.DiskEncryption, role models.HostRole) bool {
+	switch swag.StringValue(encryption.EnableOn) {
 	case models.DiskEncryptionEnableOnAll:
 		return true
 	case models.DiskEncryptionEnableOnMasters:
-		return c.host.Role == models.HostRoleMaster || c.host.Role == models.HostRoleBootstrap || c.host.Role == models.HostRoleAutoAssign
+		return role == models.HostRoleMaster || role == models.HostRoleBootstrap
 	case models.DiskEncryptionEnableOnWorkers:
-		return c.host.Role == models.HostRoleWorker || c.host.Role == models.HostRoleAutoAssign
-	default: // models.DiskEncryptionEnableOnNone
+		return role == models.HostRoleWorker
+	default:
 		return false
 	}
 }
 
-func isAutoAassignAllowed(c *validationContext) bool {
-	return *c.cluster.DiskEncryption.EnableOn == models.DiskEncryptionEnableOnAll ||
-		*c.cluster.DiskEncryption.EnableOn == models.DiskEncryptionEnableOnNone
-}
-
 func (v *validator) diskEncryptionRequirementsSatisfied(c *validationContext) ValidationStatus {
 
-	if c.infraEnv != nil || reflect.DeepEqual(c.cluster.DiskEncryption, &models.DiskEncryption{}) || !isDiskEncryptionRelevantForHost(c) {
+	if c.infraEnv != nil || reflect.DeepEqual(c.cluster.DiskEncryption, &models.DiskEncryption{}) {
 		return ValidationSuccessSuppressOutput
 	}
 
-	if c.host.Role == models.HostRoleAutoAssign && !isAutoAassignAllowed(c) {
-		return ValidationFailure
+	if c.inventory == nil {
+		return ValidationPending
 	}
 
-	if c.inventory == nil {
+	role := common.GetEffectiveRole(c.host)
```

Comment:
Why don't you check directly `c.host.Role` ?

Reply:
Because it has to take into account suggested role as well